server/service/service_users.go

```diff
@@ -34,13 +34,24 @@ func (svc *Service) newUser(ctx context.Context, p fleet.UserPayload) (*fleet.Us
 	var ssoEnabled bool
 	// if user is SSO generate a fake password
 	if (p.SSOInvite != nil && *p.SSOInvite) || (p.SSOEnabled != nil && *p.SSOEnabled) {
-		fakePassword, err := server.GenerateRandomText(14)
-		if err != nil {
-			return nil, ctxerr.Wrap(ctx, err, "generate stand-in password")
+		fakePassword := ""
+		for i := 1; i < 10; i++ {
+			// more than one try may be needed to generate a stand-in password that meets enhanced
+			// security requirements (two tries were sufficient in over 99% of tests)
+			pwd, err := server.GenerateRandomText(14)
+			if err != nil {
+				return nil, ctxerr.Wrap(ctx, err, "generate stand-in password")
+			}
+			if err = fleet.ValidatePasswordRequirements(pwd); err == nil {
+				fakePassword = pwd
+				break
+			}
```

Comment:
@lucasmrod, if I understand correctly, we would rely solely on `GenerateRandomText` for the fake password (which could result in a stored password that doesn't meet the complexity requirements in a relatively small percentage of cases (approximately 4%). Do you know how the fake/stored password is used in other contexts? I'm not very clear on how/why it exists apart from ensuring the db value is not null for SSO users. I presumed it must be something more than that if we going through the trouble of generating random text value but maybe that is all there is to it and the complexity check isn't adding any security here.

Reply:
> if I understand correctly, we would rely solely on GenerateRandomText for the fake password (which could result in a stored password that doesn't meet the complexity requirements in a relatively small percentage of cases (approximately 4%).

IMO this is probably ok as it is random text generated by `crypto/rand.Read`.

> Do you know how the fake/stored password is used in other contexts? I'm not very clear on how/why it exists apart from ensuring the db value is not null for SSO users. I presumed it must be something more than that if we going through the trouble of generating random text value but maybe that is all there is to it and the complexity check isn't adding any security here.

Asked myself the same and reached the same conclusion (it's there only due to the `NOT NULL` constraint in `password`). And `GenerateRandomText` is safer than setting an empty or fixed string (in case of SSO vs non-SSO logic bugs).